crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -574,6 +575,10 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
                     .wait_for_checkpoint_catchup(highest_checkpoint, Duration::from_secs(30))
                     .await;
 
+                if force_objects_snapshot {
```

Comment:
nit: This should be its own dedicated command (like `create-checkpoint` etc) rather than a flag on `run-graphql`.

Reply:
My decision to include it as a flag on run-graphql is because create-checkpoint is a command that tells Simulacrum to simulate new checkpoints, but does not communicate to Indexer to commit those checkpoints. Instead, the latter service asynchronously commits them. To reduce flakiness, the RunGraphqlCommand pulls the latest checkpoint sequence number from executor (and simulacrum) and waits for indexer to catch up. In a similar vein, force-objects-snapshot waits for indexer to catch up to a consistent state that we can test.

Instead of running it all the time as we do with waiting for checkpoint catchup, I chose to include it as a flag in RunGraphqlCommand, since it usually is not significant, unless you're explicitly trying to test behavior around this.

If we'd like, perhaps these are better suited in an IndexerCommand to control indexer's behavior around indexing the fullnode.